Include "Additional Fields" in CSV import

zacwasielewski opened the pull request:
This branch allows the inclusion of [Additional Fields](https://theeventscalendar.com/knowledgebase/pro-additional-fields/) (a PRO feature) in CSV imports.

This seems to be a popular feature request:

- https://theeventscalendar.com/support/forums/topic/import-to-custom-field/
- https://theeventscalendar.com/support/forums/topic/additional-fields-in-csv-import/
- https://theeventscalendar.com/support/forums/topic/mapping-additional-fields-for-csv-import/
- https://theeventscalendar.com/support/forums/topic/csv-import-to-additional-fields/
- https://theeventscalendar.com/support/forums/topic/importing-a-csv-with-custom-fields/
- https://theeventscalendar.com/support/forums/topic/making-additional-fields-items-available-to-the-csv-import-function/
- http://tribe.uservoice.com/forums/195723-feature-ideas/suggestions/5076620-add-custom-fields-to-csv-importer

bordoni commented:
Hi @zacwasielewski, I will add some comments on your code, if you can fix these we can Merge the Pull Request. :wink:

bordoni commented on src/Tribe/API.php:
```diff
@@ -270,6 +270,40 @@ public static function update_event_cost( $event_id ) {
+		 */
+		public static function update_additional_fields ( $event_id, $data ) {
+			$additionalFieldTags = array_map(
+				function( $field ) { return $field['name']; },
```
I would love for you to use a structure for variables like the following:
E.g.: `$additional_field_tags`

bordoni replied:
We cannot use Anonymous because it is not Supported by PHP 5.2, a good solution in this case is to use [`wp_list_pluck`](https://codex.wordpress.org/Function_Reference/wp_list_pluck)

bordoni commented on src/Tribe/Importer/Column_Mapper.php:
```diff
@@ -88,4 +89,12 @@ private function get_organizer_column_names() {
+	private function get_additional_fields_column_names() {
+		$additionalFields = tribe_get_option( 'custom-fields' );
+		return array_combine(
+			array_map(function($field) { return $field['name']; }, $additionalFields),
```
Same thing here with the Anonymous function, on the first one you should use `wp_list_pluck`.

On the second one, you could `sprintf( __( 'Event %s', 'tribe-events-calendar' ), $field['label'] );` but create a method on the class to do it.

bordoni commented on src/Tribe/Importer/File_Importer_Events.php:
```diff
@@ -160,6 +160,12 @@ private function build_event_array( array $record ) {
+		$additionalFields = tribe_get_option( 'custom-fields' );
+		foreach ( $additionalFields as $field ) {
+			$key = $field['name'];
+			$event[$key] = $this->get_value_by_key( $record, $key );
```
Check this on all the changes you applied, we normally use the following structure for variable `array` index:
```
$event[ $key ] = ...
```

zacwasielewski commented:
Thanks for all the input! I was away for the weekend, so just seeing these – I'll make the changes as you recommended.

zacwasielewski commented:
@bordoni I finished the edits you recommended, but just realized I was working against the `release/121` branch instead of `develop`.

I'm re-applying my commits to `develop` now – does it make sense for me to close this pull request and create a new one?

bordoni commented:
Nahh you did the right thing. Keep working on `release/121` we are using it as the main branch now. We gotta update the docks.

zacwasielewski commented:
Thanks! While merging, I noticed that @jazbek already did some work related to this feature: https://github.com/moderntribe/the-events-calendar/pull/232

Her code adds a couple of new filters that are totally relevant here, so I'm reworking my code to play nicely with them.

zacwasielewski commented:
@bordoni I cleaned up my edits as you mentioned, and also added integration with the `tribe_events_csv_import_event_additional_fields` hook in the pull request above. Any other thoughts?

zbtirrell commented:
@zacwasielewski This is an amazing contribution! Unfortunately, the bad news is that @jazbek has been working on this exact functionality in our Events Pro repository for our next release (`release/120`).  I think she matched the same functionality you've provided.

On the good side, this functionality will be shipping real soon!  Also, I went ahead and added you to our "Community Contributors" group which extends your access to our Events Pro repository.  

I want to be sure that you know we really appreciate contributions like yours and also apologize for the circumstance where duplicate work was happening.

zacwasielewski commented:
Thanks for the gesture, @zbtirrell, I appreciate the consideration. And no apologies necessary – I'm just happy the functionality is going to ship. (I built the feature myself because I urgently needed it for a project, then thought it would be useful to share.)

And after reviewing further, I agree that this code really does belong in Events Pro.

Thanks for maintaining this awesome project – I'll continue trying to contribute where I can!
